File: java/src/main/java/gov/doc/ntia/sigmf/examples/algorithm/TimeDomainDetectionExample.java
package gov.doc.ntia.sigmf.examples.algorithm;

import gov.doc.ntia.sigmf.Annotation;
import gov.doc.ntia.sigmf.Global;
import gov.doc.ntia.sigmf.MetaDoc;
import gov.doc.ntia.sigmf.examples.Example;
import gov.doc.ntia.sigmf.examples.ExampleUtils;
import gov.doc.ntia.sigmf.ext.annotation.algorithm.TimeDomainDetection;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.List;

public class TimeDomainDetectionExample implements Example {

  @Override
  public MetaDoc getExample() {

    TimeDomainDetection tdd = new TimeDomainDetection();
    tdd.setNumberOfSamples(1024L);
    tdd.setSampleStart(0L);
    tdd.setDetector("mean");
    tdd.setSampleCount(1024L);
    tdd.setUnits("dBm");
    tdd.setReference("antenna output");
    List<Annotation> annotations = new ArrayList();
    annotations.add(tdd);
    MetaDoc metaDoc = new MetaDoc();
    Global global = ExampleUtils.getGlobal();
    metaDoc.setGlobal(global);
    metaDoc.setAnnotations(annotations);
    return metaDoc;
  }
}

File: Csharp/sigmf/sigmf/examples/algorithm/TimeDomainDetectionExample.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace sigmf.examples.algorithm
{
    public class TimeDomainDetectionExample : Example
    {
        public override MetaDoc getExample() {
            MetaDoc metaDoc = new MetaDoc();
            Globals globals = ExampleUtils.getGlobals(getCurrentTime());
            List<Annotation> annotations = new List<Annotation>();

            TimeDomainDetection tdd = new TimeDomainDetection();
            tdd.NumberOfSamples = 10241;
            tdd.SampleStart = 01;
            tdd.Detector = "mean";
            tdd.DetectionDomain = "time";
            tdd.SampleCount = 10241;
            tdd.Units = "dBm";
            tdd.Reference = "antenna output";
            annotations.Add(tdd);
            metaDoc.setGlobal(globals);
            metaDoc.setAnnotations(annotations);
            return metaDoc;

        }
        private DateTime getCurrentTime()
        {
            DateTime st = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
            TimeSpan t = (DateTime.Now.ToUniversalTime() - st);
            return st;
        }
    }
}
